### PR TITLE
Revive pull request #30

### PR DESCRIPTION
**Pull request by jacobi007**

### CODE REVIEW

**Comment by ABaldwinHunter** [comment]
@jacobi007 Thanks for your work! Someone from our team will review this PR soon.

**Comment by brynary** [comment]
@jacobi007 I think I fixed the build issue in master. Can you pleasemerge master in here and see if it builds green?

**Comment by jacobi007** [comment]
@brynary it didn't help :/

**Comment by brynary** [comment]
Have another idea. Working on it.

**Comment by brynary** [comment]
OK, can you please try one more time?

**Comment by brynary** [comment]
OK, tried another thing. Another merge please? :)

File: lib/cc/cli/analyze.rb [hunk 1]
```diff
@@ -43,7 +43,7 @@ def registry
       end
 
       def formatter
-        @formatter ||= Formatters::PlainTextFormatter.new
+        @formatter ||= Formatters::PlainTextFormatter.new(filesystem)
```
Comment (brynary) [inline lib/cc/cli/analyze.rb]: We need to adjust the usage of `Formatters.resolve` so it also passes a `filesystem` param

File: lib/cc/analyzer/formatters/formatter.rb [hunk 2]
```diff
@@ -2,7 +2,8 @@ module CC
   module Analyzer
     module Formatters
       class Formatter
-        def initialize(output = $stdout)
+        def initialize(filesystem, output = $stdout)
```
Comment (brynary) [inline lib/cc/analyzer/formatters/formatter.rb]: The `initialize` method in the `JSONFormatter` will need adjustment to accept `filesystem` (currently takes no args)

**Comment by brynary** [comment]
Build fixed. There's a couple (pre-existing) bugs that need to be resolved, however.

**Comment by brynary** [comment]
(Added line comments about said bugs)

**Comment by jacobi007** [comment]
Noted, thanks!

**Comment by jacobi007** [comment]
I've modified the JSONFormatter and it takes a filesystem argument now. On the other hand I'm wondering if that's the best solution because the variable remains unused.

**Comment by brynary** [comment]
@jacobi007 looks good to me.

We'd eventually need the `filesystem` object if we wanted to make the JSON data more closely match the plain text output, but I think it's ok like this for now.

**Pull request merged**